src/com/google/javascript/jscomp/ScopedAliases.java

```diff
@@ -658,26 +658,36 @@ public void visit(NodeTraversal t, Node n, Node parent) {
         }
       }
 
-      // Validate all descendent scopes of the goog.scope block.
-      if (inGoogScopeBody()) {
-        // Check if this name points to an alias.
-        if (aliasVar != null) {
-          // Note, to support the transitive case, it's important we don't
-          // clone aliasedNode here.  For example,
-          // var g = goog; var d = g.dom; d.createElement('DIV');
-          // The node in aliasedNode (which is "g") will be replaced in the
-          // changes pass above with "goog".  If we cloned here, we'd end up
-          // with <code>g.dom.createElement('DIV')</code>.
-          aliasUsages.add(new AliasedNode(aliasVar, n));
-        }
+      // If this is a bleeding function expression, like
+      // var x = function y() { ... }
+      // then old versions of IE declare "y" in the current scope. We don't
+      // want the scope unboxing to add "y" to the global scope, so we
+      // need to rename it.
+      //
+      // The simplest solution is to add "y" to forbiddenLocals so it will get
+      // renamed.
+      if (NodeUtil.isBleedingFunctionName(n)) {
+        forbiddenLocals.add(n.getString());
```

Comment:
Looks like this is going to introduce a lot of artificial "namespace shadow"s, which we currently don't handle very well. I've hit a failure case that looks like:
```js
/** @param {?} g */
function f(g) {}
use(function g() {})
```
Since `g` is a bleeding function name, it will be renamed to `g$jscomp$1` and the code will look like:
```js
/** @param {?} g */
function f(g$jscomp$1) {}
use(function g$jscomp$1() {})
```
which will fail typechecking because the parameter is only renamed in code, but not in the JsDoc.